src/error.rs

```diff
@@ -95,8 +95,10 @@ pub enum RevisionParseError {
 }
 
 /// Errors encountered when parsing a [`Constraint`](crate::Constraint) from a string.
-#[derive(Error, Clone, PartialEq, Eq, Debug)]
```

Comment:
`semver::Error` does not implement these traits, so I removed them. Don't have a strong opinion on this, so if you think it's better to drop that error (or convert it to a String or something) so as to keep these traits, that's fine with me.

Reply:
hmmm, let's convert it to a string please. I think these are valuable to keep if we can.

Reply:
actually, thinking about this more, never mind. I'm good to leave it as is.